app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -141,6 +141,36 @@ def bulk_cancel
         }
       end
 
+      api :POST, '/tasks/bulk_stop', N_('Stop all stoppable tasks')
+      param :search, String, :desc => N_('Stop tasks matching search string')
+      param :task_ids, Array, :desc => N_('Stop specific tasks by ID')
+      def bulk_stop
+        if params[:search].nil? && params[:task_ids].nil?
+          raise BadRequest, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+
+        filtered_scope = bulk_scope
+        total_length = filtered_scope.count
+        to_stop = filtered_scope.where.not(state: :stopped)
+        to_stop_length = to_stop.count
+        skipped_length = total_length - to_stop_length
+
+        to_stop.find_each do |task|
+          task.update(state: :stopped)
+        end
```

Comment:
By the curly braces I meant only the ones inside the parentheses. There is not reason to spread this over several lines